<?xml version="1.0"?>
<launch>

  <include file="$(find navio2_ros)/launch/imu_node.launch" />

  <!-- Transforms imu data in imu_link frame to ENU data in base_link frame -->
  <!-- <node name="imu_data_transformer" pkg="imu_transformer" type="imu_transformer_node" >
    <remap from="imu_in/data" to="imu_lsm/data_raw" />
    <remap from="imu_in/mag" to="imu_lsm/mag" />
    <remap from="imu_out/data" to="imu/data_raw" />
    <remap from="imu_out/mag" to="imu/mag" />
    <param name="target_frame" value="base_link" />
  </node> -->
  
  <!-- Filter raw gyro, accel and mag data into a usable orientation -->
  <node name="imu_filter" pkg="imu_filter_madgwick" type="imu_filter_node" >
    <rosparam command="load" file="$(find navio2_ros)/config/imu_calibration.yaml" />
    <remap from="/imu/data_raw" to="/imu_lsm/data_raw" />
    <remap from="/imu/mag" to="/imu_lsm/mag" />
    <remap from="/imu/data" to="/imu/data" />
    <param name="orientation_stddev" value="0.001" />
    <param name="gain" value="0.01" />
    <param name="zeta" value="0.001" />
    <param name="publish_tf" value="false" />
  </node>
</launch>
